cmd/gateway.go

```diff
@@ -55,25 +62,37 @@ var gatewayCmd = &cobra.Command{
 		// Set up HTTP handler
 		http.Handle("/", managerInstance)
 
-		// Start HTTP server
-		err = http.ListenAndServe(fmt.Sprintf(":%s", appCfg.Port), nil)
-		if err != nil {
-			log.Error().Err(err).Msg("Error starting server")
-			return fmt.Errorf("failed to start server: %w", err)
+		// Start HTTP server with context
+		server := &http.Server{
+			Addr:    fmt.Sprintf(":%s", appCfg.Gateway.Port),
+			Handler: nil,
 		}
 
-		log.Info().Float64("elapsed_seconds", time.Since(start).Seconds()).Msg("Setup completed")
+		// Start the HTTP server in a goroutine so that we can listen for shutdown signals
+		go func() {
+			err := server.ListenAndServe()
+			if err != nil && !errors.Is(err, http.ErrServerClosed) {
+				log.Error().Err(err).Msg("Error starting HTTP server")
+			}
+		}()
+
+		// Wait for shutdown signal via the context
+		<-ctx.Done()
+
+		// Perform graceful shutdown of HTTP server
+		log.Info().Msg("Shutting down HTTP server...")
+		if err := server.Shutdown(ctx); err != nil {
```

Comment:
You can not use the already closed `ctx` here. You need to create a new context (with a timeout maybe).

Reply:
Good catch, I created a new context with timeout from the default config.